File: 0223-Rectangle Area/0223-Rectangle Area.cpp
class Solution {
public:
    int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        int left = max(A, E);
        int right = max(min(C, G), left);
        int bottom = max(B, F);
        int up = max(min(D, H), bottom);
        return (C - A) * (D - B) + (G - E) * (H - F) - (right - left) * (up - bottom);
    }
};

File: 0201-0300/0223-Rectangle Area/0223-Rectangle Area.py
class Solution:
    def computeArea(self, A: int, B: int, C: int, D: int, E: int, F: int, G: int, H: int) -> int:
        left = max(A, E)
        right = max(min(C, G), left)
        bottom = max(B, F)
        up = max(min(D, H), bottom)
        return (C - A)*(D - B) + (G - E)*(H - F) - (right - left)*(up - bottom)
